### DISCUSSION TITLE
How does wait method works

### DISCUSSION THREAD

**Discussion by dmurashko-ms** [faaxm/spix#120] [General]
It might be my mistake.
But when I use wait(10 * 1000) to wait 10 seconds nothing is blocked(script or application).
Or it's sleep for Rpc server?

**Comment by faaxm** [comment]
Hi @dmurashko-ms,
the wait only relates to the time the next event is fired.
So, if you do a "mouse-click, wait 10s, mouse-click", then the second mouse-click will only be performed at least 10s after the first click. Everything else will keep running.

You could use a "generic command handler" (https://github.com/faaxm/spix#using-genericcustom-command) with some cusom code if you want to block the whole application.

**Reply by dmurashko-ms** [reply]
Thanks for the reply
